### PR TITLE
masked std

### PR DESCRIPTION
**Pull request by george-qi**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72723
* __->__ #72722

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1070,6 +1070,10 @@ def sample_inputs_masked_var(op_info, device, dtype, requires_grad, **kwargs):
                     # autograd output that test_grad_fn cannot handle
                     # correctly.
                     continue
+                if sample_input.input.dim() == 0 or orig_count.min() == 1:
+                    # Skip samples when the autograd output for std could not be handled correctly
+                    # due to torch.sqrt
+                    continue
```
Comment (pearu) [inline torch/testing/_internal/common_methods_invocations.py:1076]: This change is unnecessary if one corrects
https://github.com/pytorch/pytorch/blob/1f29b3130af218847a043e58fdc64511bbe072fe/torch/testing/_internal/common_methods_invocations.py#L1068
to
```python
        if orig_count.min() <= int(unbiased) + 1:
```
```suggestion
```

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -15177,8 +15181,40 @@ def ref_pairwise_distance(input1, input2):
             DecorateInfo(toleranceOverride({torch.float16: tol(atol=1e-02, rtol=1e-02)}),
                          'TestMasked', 'test_reference_masked'),
         ],
-        sample_inputs_func=sample_inputs_masked_var,
-        gradcheck_wrapper=gradcheck_wrapper_masked_operation
+        sample_inputs_func=sample_inputs_masked_std_var,
+        gradcheck_wrapper=gradcheck_wrapper_masked_operation,
+        check_batched_grad=False,
+        check_batched_forward_grad=False,
```
Comment (pearu) [inline torch/testing/_internal/common_methods_invocations.py:15187]: See the correction at line 1068 above.
```suggestion
```

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -15177,8 +15181,40 @@ def ref_pairwise_distance(input1, input2):
...
+                         'TestReductions', 'test_reference_masked'),
+            DecorateInfo(toleranceOverride({torch.float16: tol(atol=1e-02, rtol=1e-02)}),
+                         'TestReductions', 'test_ref_small_input'),
+            DecorateInfo(toleranceOverride({torch.float16: tol(atol=1e-02, rtol=1e-02)}),
+                         'TestMasked', 'test_reference_masked'),
+        ],
+        sample_inputs_func=sample_inputs_masked_std_var,
+        gradcheck_wrapper=gradcheck_wrapper_masked_operation,
+        check_batched_grad=False,
+        check_batched_forward_grad=False,
```
Comment (pearu) [inline torch/testing/_internal/common_methods_invocations.py:15217]: See the correction at line 1068 above.
```suggestion
```

**Review by pearu — CHANGES_REQUESTED** [review]
Looks good, thanks, @george-qi !

I have a suggestion to fix a bug in `sample_inputs_masked_std_var` function so that `batched_grad` and `batched_forward_grad` checks can be re-enabled for masked `var` and `std` operations.
